File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if(k==0) return 0;
        int low=0,high=0, count=0;
        long long products = 1; 
        while(high<nums.size())
        {
            products*=nums[high];

            while(products>=k && low<=high){
                products/=nums[low];
                low++;
            }
            count+=high-low+1;
            high++;
        }
        return count;
    }
};

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k<=1: 
            return 0
        
        prod = 1
        right, left = 0,0
        ans = 0

        while(right<len(nums)):
            prod*=nums[right]
            while(prod>=k):
                prod //= nums[left]
                left+=1
            ans += (right-left+1)
            right+=1
        return ans